lib/msgpax.ex

```diff
@@ -134,7 +118,8 @@ defmodule Msgpax do
   a MessagePack-serialized term with nothing after that, it accepts leftover
   bytes at the end of `iodata` and only de-serializes the part of the input that
   makes sense. It returns `{:ok, term, rest}` if de-serialization is successful,
-  `{:error, reason}` otherwise.
+  `{:error, exception}` otherwise (where `exception` is a `Msgpax.UnpackError`
+  struct).
 
```

Comment:
I think it's worth mentioning because I'm not sure people are already used to going to the documentation for exceptions (since the thing of returning exceptions as structs that can be raised/logged is a quite "new" convention in Elixirland). I would keep it in the other functions. I can add it here, but we have to redirect to `unpack/2` anyways for possible errors, options and so on so I think it can be okay to be more concise here.